src/status_im2/subs/communities.cljs

```diff
@@ -223,31 +207,62 @@
         (sort-by :position)
         (into []))))
 
+(defn reduce-over-categories
+  [community-id
+   joined
+   categories
+   collapsed-categories
+   full-chats-data]
+  (fn [acc
+       [_ {:keys [name categoryID position id emoji can-post?]}]]
+    (let [category-id                       categoryID
+          {:keys [unviewed-messages-count
+                  unviewed-mentions-count]} (get full-chats-data
+                                                 (str community-id id))
+          acc-with-category                 (if (get acc category-id)
+                                              acc
+                                              (assoc acc
+                                                     category-id
+                                                     (assoc
+                                                      (or (get categories category-id)
+                                                          {:name (i18n/label :t/none)})
```

Comment:
non-categorized chats go into the `nil` key

Reply:
Many things can be nil in Clojure, I think a more robust and explicit approach is to store in a qualified keyword such as `:communities/non-categorized` or any other name really, as long as it's guaranteed to be a unique key in the map.

Reply:
in theory, it's always possible that the user would call a category -> `communities/non-categorized`
but the distinction would be that one is a keyword and one is a string, but that also looks a bit fragile, since the temptation to store keywordize keys for devs is too tempting :)

Reply:
Good point, we could end up getting the user's data if the code didn't use a keyword and used the stringified key by mistake. It's worth a comment, this is not obvious since it's very rare in Clojure to use `nil` as a keyword, first time to me actually.

One reason we don't see this practice in Clojure is because there's a very high chance that somebody will call `get` on the map and take advantage of nil punning to pipe the result to something else, but because the `nil` key is actually valid, developers would need to always differentiate between the default value and the key not found all the time, which is quite against the grain.

Also it looks very weird to read code like this: `(get m nil)`. But anyway @cammellos, you brought a good point. I don't have any better idea now.

Reply:
I think probably the safest is just as you suggested `:communities/not-categorized`, the chances of someone hitting that are minimal, thanks!

Reply:
I'm going to use it for my community 😈

Reply:
Just sharing here, Andrea and I were talking off GH now about some weird things related to destructuring maps, and we realized we can use symbols as keys in Clojure (also one of the most common practices in other Lisps). So we can really store something unique in the map that can never conflict with user data o/